Recurse.ps1

```diff
@@ -1,15 +1,86 @@
+    exit
+}
+
+# Check if GitHub CLI is installed
+try {
+    gh | Out-Null
+} catch [System.Management.Automation.CommandNotFoundException] {
+    Write-Host "GitHub CLI is not installed." -ForegroundColor Red
+    exit
+}
+
+# If the user has git installed, make sure it is a patched version
+if (Get-Command 'git.exe' -ErrorAction SilentlyContinue) {
+    $GitMinimumVersion = [System.Version]::Parse('2.35.2')
+    $gitVersionString = ((git version) | Select-String '([0-9]{1,}\.){3,4}').Matches.Value.Trim(' ', '.')
+    $gitVersion = [System.Version]::Parse($gitVersionString)
+    if ($gitVersion -lt $GitMinimumVersion) {
+        # Prompt user to install git
+        if (Get-Command 'winget.exe' -ErrorAction SilentlyContinue) {
+            $_menu = @{
+                entries       = @('[Y] Upgrade Git'; '[N] Do not upgrade')
+                Prompt        = 'The version of git installed on your machine does not satisfy the requirement of version >= 2.35.2; Would you like to upgrade?'
+                HelpText      = "Upgrading will attempt to upgrade git using winget`n"
+                DefaultString = ''
+            }
+            switch (Invoke-KeypressMenu -Prompt $_menu['Prompt'] -Entries $_menu['Entries'] -DefaultString $_menu['DefaultString'] -HelpText $_menu['HelpText']) {
+                'Y' { 
+                    Write-Host
+                    try {
+                        winget upgrade --id Git.Git --exact
+                    } catch {
+                        throw [UnmetDependencyException]::new('Git could not be upgraded sucessfully', $_)
+                    } finally {
+                        $gitVersionString = ((git version) | Select-String '([0-9]{1,}\.){3,4}').Matches.Value.Trim(' ', '.')
+                        $gitVersion = [System.Version]::Parse($gitVersionString)
+                        if ($gitVersion -lt $GitMinimumVersion) {
+                            throw [UnmetDependencyException]::new('Git could not be upgraded sucessfully')
+                        }
+                    }
+                 }
+                default { Write-Host; throw [UnmetDependencyException]::new('The version of git installed on your machine does not satisfy the requirement of version >= 2.35.2') }
+            }
+        } else {
+            throw [UnmetDependencyException]::new('The version of git installed on your machine does not satisfy the requirement of version >= 2.35.2')
+        }
+    }
+    # Check whether the script is present inside a fork/clone of jedieaston/Add-ARPEntries repository
+    try {
+        $script:gitTopLevel = (Resolve-Path $(git rev-parse --show-toplevel)).Path
+    } catch {
+        # If there was an exception, the user isn't in a git repo. Throw a custom exception and pass the original exception as an InternalException
+        throw [UnmetDependencyException]::new('This script must be run from inside a clone of the Add-ARPEntries repository', $_.Exception)
```

Comment:
Suggestion: we should make it so you can just have Add-ARPEntries.ps1 on your PATH, and it can use it (so we don't have to necessarily run this from right next to the script).